prow/pubsub/subscriber/subscriber.go

```diff
@@ -58,6 +61,90 @@ type prowCfgClient interface {
 	GetPostsubmitsStatic(identifier string) []config.Postsubmit
 }
 
+type InRepoConfigCacheGetter struct {
+	CookieFilePath string
+	CacheSize      int
+	Agent          *config.Agent
+	mu             sync.Mutex
+	GithubOptions  flagutil.GitHubOptions
+	DryRun         bool
+
+	CacheMap map[string]*config.InRepoConfigCache
+}
+
+func (irc *InRepoConfigCacheGetter) GetCache(cloneURI, host string) (*config.InRepoConfigCache, error) {
+	irc.mu.Lock()
+	defer irc.mu.Unlock()
+
+	var cache *config.InRepoConfigCache
+	var gitClientFactory git.ClientFactory
+
+	// Are we using github with InRepoConfig
+	if irc.GithubOptions.TokenPath != "" || irc.GithubOptions.AppPrivateKeyPath != "" {
```

Comment:
the code in this block has very little difference from `CreateGerritCache`, I think they can be combined

Reply:
Done